### PR TITLE
fix PublishedNodesConverter

### PR DESCRIPTION
**Pull request by cristipogacean**
handle the events handling in the publish nodes converter

### CODE REVIEW

**Comment by dacolgit** [comment]
> 

remove empty line

---
Refers to: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Core/Extensions/SimpleAttributeOperandModelEx.cs:64 in c6335af. [](commit_id = c6335af0927b5624c9a558952d4cac44a81920f5, deletion_comment = False)

**Comment by dacolgit** [comment]
> 

remove empty line

---
Refers to: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/PublishedDataSetSourceModelEx.cs:69 in c6335af. [](commit_id = c6335af0927b5624c9a558952d4cac44a81920f5, deletion_comment = False)

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/DataFilterModel.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// Describing an event entry in the configuration.
+    /// </summary>
+    [DataContract]
+    public class DataFilterModel {
+
+        /// <summary>
+        /// Simple event Type id
```
Comment (dacolgit) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/DataFilterModel.cs]: > Simple event Type id [](http://example.com/codeflow?start=11&length=21)

fix comment

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/DataFilterModel.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.OpcUa.Publisher.Config.Models {
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// Describing an event entry in the configuration.
```
Comment (dacolgit) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/DataFilterModel.cs]: >Describing an event entry in the configuration. [](http://example.com/codeflow?start=8&length=47)

fix comment

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/DataFilterModel.cs (hunk 1 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/DataFilterModel.cs]: done

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/DataFilterModel.cs (hunk 2 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/DataFilterModel.cs]: done

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/BasicIntegrationTests.cs [hunk 3]
```diff
@@ -27,7 +27,8 @@ public class BasicIntegrationTests : PublisherIntegrationTestBase, IClassFixture
         }
 
         // ToDo: Enable the test once PublishedNodesJobConverter parses OpcEvents.
```
Comment (dacolgit) [inline modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/BasicIntegrationTests.cs]: >// ToDo: Enable the test once PublishedNodesJobConverter parses OpcEvents. [](http://example.com/codeflow?start=8&length=74)

remove the To Do comment

**Review by dacolgit — APPROVED** [review]
:shipit:

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/OpcNodeModel.cs [hunk 4]
```diff
@@ -60,5 +60,13 @@ public class OpcNodeModel {
         /// <summary> Queue Size for the monitored item </summary>
         [DataMember(EmitDefaultValue = false, IsRequired = false)]
         public uint? QueueSize { get; set; }
+
+        /// <summary>Event Filter </summary>
+        [DataMember(EmitDefaultValue = false, IsRequired = false)]
+        public EventFilterModel EventFilter { get; set; }
+
+        /// <summary>Event Filter </summary>
```
Comment (karok2m) [inline components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/OpcNodeModel.cs]: Data filter, maybe ?

File: modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/BasicIntegrationTests.cs (hunk 3 above)
Comment (cristipogacean) [reply modules/src/Microsoft.Azure.IIoT.Modules.OpcUa.Publisher/tests/BasicIntegrationTests.cs]: done

File: components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/OpcNodeModel.cs (hunk 4 above)
Comment (cristipogacean) [reply components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Models/OpcNodeModel.cs]: done

**Review by karok2m — APPROVED** [review]

**Comment by karok2m** [comment]
@cristipogacean, there might be some changes necessary for the E2E test parts as well to make sure that OPC Publisher configuration is generated properly. Should this be addressed in this PR or separately ?

**Comment by cristipogacean** [comment]
> @cristipogacean, there might be some changes necessary for the E2E test parts as well to make sure that OPC Publisher configuration is generated properly. Should this be addressed in this PR or separately ?

@karok2m let's address this in a separate PR.

**Pull request merged**